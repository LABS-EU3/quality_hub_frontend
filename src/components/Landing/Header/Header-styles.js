import styled from 'styled-components';
import devices from '../../devices';

export const HeaderContainer = styled.div`
  position: absolute;
  top: 20rem;
  max-width: 50rem;
  margin-left: 4.3rem;

  @media ${devices.tablet} {
    top: 16rem;
    margin: 0;
    text-align: center;
    padding: 0 2rem;
  }

  .cta-title {
    font-size: 3rem;
    font-weight: bold;
    color: white;
    width: 100%;

    @media ${devices.mobile} {
      font-size: 2.2rem;
    }
  }

  .cta-button {
    width: 15rem;
    margin-top: 2rem;

    @media ${devices.tablet} {
      margin: 2rem auto;
    }
  }
`;
