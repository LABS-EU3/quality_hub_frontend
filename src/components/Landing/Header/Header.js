import React from 'react';
import { Link } from 'react-router-dom';
import { StyledButton, buttonTheme } from '../Landing-styles';
import { HeaderContainer } from './Header-styles';

const Header = () => (
  <HeaderContainer>
    <h2 data-testid='cta-title' className='cta-title'>
      Interviewing isn't easy. That doesn't mean it has to be
      stressful
    </h2>
    <div className='cta-button'>
      <Link data-testid='cta-button' to='/register'>
        <StyledButton theme={buttonTheme}>SIGN UP</StyledButton>
      </Link>
    </div>
  </HeaderContainer>
);

export default Header;
