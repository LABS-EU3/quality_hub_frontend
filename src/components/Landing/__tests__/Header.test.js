import React from 'react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import * as rtl from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from '../Header/Header';

describe('Header', () => {
  it('matches snapshot ', () => {
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Router>
        <Header />
      </Router>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('H2 copy displayed', () => {
    const tree = rtl.render(
      <Router>
        <Header />
      </Router>,
    );
    const ctaTitle = tree.getByTestId('cta-title');

    expect(ctaTitle).toBeInTheDocument();
  });

  it('CTA Button link ok', () => {
    const tree = rtl.render(
      <Router>
        <Header />
      </Router>,
    );
    const ctaButton = tree.getByTestId('cta-button');
    expect(ctaButton).toBeInTheDocument();

    expect(ctaButton).toHaveAttribute('href', '/register');
  });
});
