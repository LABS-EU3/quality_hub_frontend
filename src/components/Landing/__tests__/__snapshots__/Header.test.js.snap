// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot  1`] = `
<div
  className="sc-bxivhb jbynth"
>
  <h2
    className="cta-title"
    data-testid="cta-title"
  >
    Interviewing isn't easy. That doesn't mean it has to be stressful
  </h2>
  <div
    className="cta-button"
  >
    <a
      data-testid="cta-button"
      href="/register"
      onClick={[Function]}
    >
      <button
        className="sc-htpNat bypFkS"
      >
        SIGN UP
      </button>
    </a>
  </div>
</div>
`;
